src/flashbar/index.tsx

```diff
@@ -17,32 +17,59 @@ import useFocusVisible from '../internal/hooks/focus-visible';
 import { useMergeRefs } from '../internal/hooks/use-merge-refs';
 import { useReducedMotion, useVisualRefresh } from '../internal/hooks/use-visual-mode';
 import { getVisualContextClassname } from '../internal/components/visual-context';
+
 import styles from './styles.css.js';
 
 export { FlashbarProps };
 
 export default function Flashbar({ items, ...restProps }: FlashbarProps) {
   const { __internalRootRef } = useBaseComponent('Flashbar');
-  const [breakpoint, ref] = useContainerBreakpoints(['xs']);
   const baseProps = getBaseProps(restProps);
-  const mergedRef = useMergeRefs(ref, __internalRootRef);
+
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [breakpoint, breakpointRef] = useContainerBreakpoints(['xs']);
+  const mergedRef = useMergeRefs(ref, breakpointRef, __internalRootRef);
+
   const isFocusVisible = useFocusVisible();
   const isVisualRefresh = useVisualRefresh();
 
+  const [previousItems, setPreviousItems] = useState<ReadonlyArray<FlashbarProps.MessageDefinition>>(items);
+  const [nextFocusId, setNextFocusId] = useState<string | null>(null);
+
+  // Track new or removed item IDs in state to only trigger focus changes for newly added items.
+  // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
+  if (items) {
+    const newItems = items.filter(({ id }) => id && !previousItems.some(item => item.id === id));
+    const removedItems = previousItems.filter(({ id }) => id && !items.some(item => item.id === id));
+    if (newItems.length > 0 || removedItems.length > 0) {
+      setPreviousItems(items);
+      const newFocusItems = newItems.filter(({ ariaRole }) => ariaRole === 'alert');
+      if (newFocusItems.length > 0) {
+        setNextFocusId(newFocusItems[0].id!);
```

Comment:
What do you think about adding a test to check if the focus order (next/prev) is working as expected?